lib/services/plugins-service.ts

```diff
@@ -77,12 +78,33 @@ export class PluginsService implements IPluginsService {
 					shelljs.cp("-R", pluginData.fullPath, pluginDestinationPath);
 				
 					let pluginPlatformsFolderPath = path.join(pluginDestinationPath, pluginData.name, "platforms", platform);
-					let pluginConfigurationFilePath = path.join(pluginPlatformsFolderPath, platformData.configurationFileName);					
+					let pluginConfigurationFilePath = path.join(pluginPlatformsFolderPath, platformData.configurationFileName);
+					let configurationFilePath = platformData.configurationFilePath;		
+					
 					if(this.$fs.exists(pluginConfigurationFilePath).wait()) {
+						// Validate plugin configuration file
 						let pluginConfigurationFileContent = this.$fs.readText(pluginConfigurationFilePath).wait();
-						let configurationFileContent = this.$fs.readText(platformData.configurationFilePath).wait();
-						let resultXml = this.mergeXml(pluginConfigurationFileContent, configurationFileContent).wait();
-						this.$fs.writeFile(platformData.configurationFilePath, resultXml).wait();	
+						
+						var doc = new DOMParser({
+							locator: {},
+							errorHandler: (level: any, msg: string) => {
+								this.$errors.fail(`Invalid xml file ${pluginConfigurationFilePath}. Additional technical information: ${msg}.` )
+							}
+						});
+						doc.parseFromString(pluginConfigurationFileContent, 'text/xml');
+						
+						// Create a backup of configuration file
+						let configurationBackupFilePath = path.join(path.dirname(configurationFilePath), "original-" + path.basename(configurationFilePath));
```

Comment:
This can potentially overwrite a user file. Consider creating the backup file in the temp directory with unique name.